trio/_path.py

```diff
@@ -182,6 +182,73 @@ async def open(self, *args, **kwargs):
         value = await trio.to_thread.run_sync(func)
         return trio.wrap_file(value)
 
+    if TYPE_CHECKING:
+        # the dunders listed in _forward_magic that aren't seen otherwise
+        __bytes__ = pathlib.Path.__bytes__
+        __truediv__ = pathlib.Path.__truediv__
+        __rtruediv__ = pathlib.Path.__rtruediv__
+
+        # Note: these are not parsed properly by jedi.
+        # It might be superior to just manually implement all the methods and get rid
+        # of all the magic wrapping stuff.
+
+        # wrapped methods handled by __getattr__
+        absolute: Any
+        as_posix: Any
+        as_uri: Any
+        chmod: Any
+        cwd: Any
+        exists: Any
+        expanduser: Any
+        glob: Any
+        home: Any
+        is_absolute: Any
+        is_block_device: Any
+        is_char_device: Any
+        is_dir: Any
+        is_fifo: Any
+        is_file: Any
+        is_reserved: Any
+        is_socket: Any
+        is_symlink: Any
+        iterdir: Any
+        joinpath: Any
+        lchmod: Any
+        lstat: Any
+        match: Any
+        mkdir: Any
+        read_bytes: Any
+        read_text: Any
+        relative_to: Any
+        rename: Any
+        replace: Any
+        resolve: Any
+        rglob: Any
+        rmdir: Any
+        samefile: Any
+        stat: Any
+        symlink_to: Any
+        touch: Any
+        unlink: Any
+        with_name: Any
+        with_suffix: Any
+        write_bytes: Any
+        write_text: Any
+
+        if sys.platform != "win32":
```

Comment:
It would be nice to be able to auto-generate this (read from relevant file in https://github.com/python/typeshed and change the types... like just stick an `async` in the right places and call it a day). I suppose that's all part of "how will we do this" that is the followup, so ignore this :^)